hicexplorer/lib/viewpoint.py

```diff
@@ -0,0 +1,188 @@
+            # index_viewpoint = 0
+            for j, idx in zip(range(elements_of_viewpoint), range(view_point_range[0], view_point_range[1], 1)):
+                # if j < view_point_start:
+                #     index_viewpoint = j
+                # elif j > view_point_end:
+                #     index_viewpoint = j
+                data_list[j] += self.hicMatrix.matrix[view_point_start_, idx]
+
+            view_point_start_ += 1
+
+        elements_of_viewpoint = elements_of_viewpoint - (view_point_end - view_point_start)
+        data_list_new = np.zeros(elements_of_viewpoint)
+
+        index_before_viewpoint = view_point_start - view_point_range[0]
+
+        # elements before the viewpoint
+        data_list_new[0:index_before_viewpoint] = data_list[0:index_before_viewpoint]
+
+        # summation because the viewpoint can be not only one bin but can contain multiple
+        data_list_new[index_before_viewpoint] = np.sum(data_list[index_before_viewpoint: index_before_viewpoint + view_point_end - view_point_start + 1])
+
+        # elements after the viewpoint
+        data_list_new[index_before_viewpoint + 1:] = data_list[index_before_viewpoint + view_point_end - view_point_start + 1:]
+        return data_list_new
+
+    def createInteractionFileData(self, pReferencePoint, pChromViewpoint, pRegion_start, pRegion_end, pInteractionData):
+
+        view_point_start, view_point_end = self.getReferencePointAsMatrixIndices(pReferencePoint)
+        view_point_range = self.getViewpointRangeAsMatrixIndices(pChromViewpoint, pRegion_start, pRegion_end)
+        elements_of_viewpoint = view_point_range[1] - view_point_range[0] - (view_point_end - view_point_start) + 1
+
+        interactions_list = []
+        chrom, start, _, _ = self.hicMatrix.getBinPos(view_point_start)
+        _, _, end, _ = self.hicMatrix.getBinPos(view_point_end)
+        interaction_positions = list(range(view_point_range[0], view_point_start, 1))
+        interaction_positions.extend([view_point_start])
+        interaction_positions.extend(list(range(view_point_end + 1, view_point_range[1], 1)))
+        relative_position = -1
+        for j, idx in zip(range(elements_of_viewpoint), interaction_positions):
+
+            chrom_second, start_second, end_second, _ = self.hicMatrix.getBinPos(idx)
+            if relative_position < 0:
+                relative_position = int(start_second) - int(start)
+            else:
+                relative_position = int(end_second) - int(end)
+
+            interactions_list.append((chrom, start, end, chrom_second, start_second, end_second, relative_position, float(pInteractionData[j])))
+
+        return interactions_list
+
+    def getViewpointRangeAsMatrixIndices(self, pChromViewpoint, pRegion_start, pRegion_end):
+
+        return self.hicMatrix.getRegionBinRange(pChromViewpoint, pRegion_start, pRegion_end)
+
+    def getReferencePointAsMatrixIndices(self, pReferencePoint):
+        if len(pReferencePoint) == 2:
```

Comment:
should pReferencePoint not be better a https://docs.python.org/2/library/collections.html#collections.namedtuple or a Bunch?